### ISSUE TITLE
Unknown USB device - Device descriptor request failed

### ISSUE THREAD

**Issue by raagz** [OSVR/OSVR-HDK-Windows-Drivers#4]
After successfully installing OSVR-HDK-Combined-Driver-Installer-1.2.4.exe and connecting devices (USB 3 connectgor from the belt base)  I get an error USB device not recognized.
In the device manager Unknown USB device - Device descriptor request failed
The same issue is showing up on Win8.1-X64 & Win10-X64 the Display is detected correctly but the tracker is not. 
Not being able to attach driver-troubleshooting-logs.txt since I don't have write pemission to this repository 

![sensicsinstall](https://cloud.githubusercontent.com/assets/15374165/10810672/8334bd56-7dc0-11e5-9483-630b9aa89cac.JPG)

![devicemanager](https://cloud.githubusercontent.com/assets/15374165/10810671/8097fcca-7dc0-11e5-9a33-570856f2b38b.JPG)

![cpuzmotherboard](https://cloud.githubusercontent.com/assets/15374165/10810670/7ca197f2-7dc0-11e5-8645-664955ca4704.JPG)

**Comment by pcmhdhp** [comment]
I also have this problem, looking for a fix.

**Comment by vt5491** [comment]
I was having this problem as well.  I had everything working fine, and then after waking my computer up from hibernation, the osvr HDK 1.3 wasn't working.  I tried plugging the usb cable in and out several times, and cycling my machine to no avail.

There are two devices that need to be attached under Device Manager: an "OSVR HDK Head-mounted Display" under "Monitors", and a generic "USB Composite Device" under "Universal Serial Bus Controllers" as the following screen print shows:

![osvr_device_screen_shot](https://cloud.githubusercontent.com/assets/8662470/12292851/b92e0cbc-b9a4-11e5-9f50-0369a57808d0.png)

I had no trouble with the HDMI connection, which shows as "OSVR HMD Display" -- it was the USB port that was giving me trouble.

In spite of the fact that the HDMI connection was OK per Device Manager, I noticed that the HMD had no lights displaying i.e. like it was turned off.  I cycled my computer one last time and then unplugged the power supply for the HMD and then re-plugged it back in.  After this, I saw the desktop on the HMD display and then everything worked fine again.

In short, try power cycling the HMD and see if this works.

**Comment by demonixis** [comment]
Hi,

I use Unity and I have this kind of problem when I load a level. In the first loaded scene all works as expected, but when I switch to another scene, the driver crashes and the HMD loose the head tracking, then the display followed by the message that the USB is not well detected.

You can see all errors in the console on the attached screenshot.

![errorosvr](https://cloud.githubusercontent.com/assets/1017126/13202637/d290d63c-d8a1-11e5-97a9-916e7d652d9e.png)

I think that this is a bug in the Unity SDK (I already opened an issue on this repo), a resource is not well disposed (or it must not be disposed?) resulting to a low level error. The driver don't catch this error and it just crashes.

**Comment by AxelLiew** [comment]
Hi I managed to fix this problem by pulling out the power cable and putting it back it. basically turn it off and turn it on again. Hope this helps.

**Comment by demonixis** [comment]
Hi @AxelLiew this is what I do to fix this problem to, but I can't say that to users that plays my games. Can we find a low level workaround to prevent that?

**Comment by rpavlik** [comment]
Has everyone participating in this discussion upgraded the firmware on their HDK to 1.85 (in the case of an HDK 1.3)?

**Comment by demonixis** [comment]
I can confirm that my HDK has the 1.85 version of the firmware.

**Comment by rpavlik** [comment]
The issue being described here may have more than one cause: it's a symptom, not a single bug or issue. The beta 1.89 firmware now available fixes all known causes of this issue - see https://github.com/OSVR/OSVR-Docs/blob/master/Utilities/OSVRControl.md to upgrade.

**Comment by demonixis** [comment]
Hi,
I have upgraded to the beta firmware but I still have some crashes.

**Comment by rpavlik** [comment]
@demonixis Rats - you must be an excellent bug finder :-P We'll be in touch.

**Comment by demonixis** [comment]
@rpavlik I do my best ;) But note that this problem happens only in direct mode with unity. I noticed that it appears sooner if I call the recenter function.

**Comment by DuFF14** [comment]
The crashes I see occur in direct or non-direct mode in the VRSamples project and some others. For me, it happens in the Editor only, when switching between scenes (ones that are not part of the plugin). Not sure if @demonixis issue is exactly the same. The argument against it being firmware related is that it also happens when testing with the DK2. I get errors about an object being used after it was disposed. The fact that it's only in certain scenes (more complex ones than the example scenes in the plugin) might indicate that it has to do with other objects in the scene that are referencing ClientKit or one of the other OSVR scripts. More info here: https://github.com/OSVR/OSVR-Unity/issues/127

**Comment by rpavlik** [comment]
so those sound unrelated - crashes vs. device descriptor failures (basically USB protocol errors). The worst thing that will happen if the HDK's USB stops working in the middle of an app should be that orientation tracking stops working until you unplug it and replug (which you can do while the server and app are running) - the server won't crash, the client shouldn't crash. The device descriptor failures won't cause crashes.
